File: boj/b_17087.js
function solution(n, s, arr) {
  const gcd = (a, b) => (a % b === 0 ? b : gcd(b, a % b));
  arr.push(s);
  arr.sort((a, b) => a - b);
  let result = arr[1] - arr[0];
  for (let i = 2; i <= n; i += 1) {
    result = gcd(result, arr[i] - arr[i - 1]);
  }
  return result;
}
console.log(solution(3, 81, [33, 105, 57]));
console.log(solution(3, 3, [1, 7, 11]));

File: boj/b_17087.py
import sys
sys.stdin = open("input.txt", "r") # 제거
input = sys.stdin.readline

def gcd_n(arr):
    def gcd(a, b):
        while b > 0:
            a, b = b, a % b
        return a
    a = arr[0]
    for v in arr:
        a = gcd(a, v)
    return a

n, s = map(int, input().split())
arr = sorted([s] + list(map(int, input().split())))
diff = []
for i in range(1, n + 1):
    diff.append(arr[i] - arr[i - 1])
print(gcd_n([1, 3, 7, 11]))

# 내장 메서드 사용
import sys, math
sys.stdin = open("input.txt", "r") # 제거
input = sys.stdin.readline

n, s = map(int, input().split())
arr = sorted([s] + list(map(int, input().split())))
diff = []
result = arr[1] - arr[0]
for i in range(2, n + 1):
    result = math.gcd(result, arr[i] - arr[i - 1])
print(result)